src/dtype.rs

```diff
@@ -94,12 +106,202 @@ impl PyArrayDescr {
         }
     }
 
-    /// Retrieves the
-    /// [enumerated type](https://numpy.org/doc/stable/reference/c-api/dtype.html#enumerated-types)
-    /// for this type descriptor.
-    pub fn get_typenum(&self) -> c_int {
+    /// Returns a unique number for each of the 21 different built-in
+    /// [enumerated types](https://numpy.org/doc/stable/reference/c-api/dtype.html#enumerated-types).
+    ///
+    /// These are roughly ordered from least-to-most precision.
+    ///
+    /// Equivalent to [`np.dtype.num`](https://numpy.org/doc/stable/reference/generated/numpy.dtype.num.html).
+    pub fn num(&self) -> c_int {
```

Comment:
This has been renamed from `get_typenum()` to stay close to `np.dtype` Python API. Very unlikely it was used anywhere (and in the internal code there was only one reference).

Reply:
Personally, I don't mind the rename seeing our version number and considering the theme of deriving the names from `numpy.dtype`.